src/libraries/System.Runtime.Caching/tests/System.Runtime.Caching.Tests.csproj

```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent)-Unix;$(NetFrameworkCurrent)</TargetFrameworks>
```

Comment:
Don't see why this had to change

Reply:
This was added bcoz System.Runtime.Caching tests currently fail for browser as we no longer support System.Diagnostics.process on wasm, so either we will have start PNSE on the System.Runtime.Caching or change the tests.

Reply:
I see, you are preventing the tests from running on Browser by splitting the confligurations to exclude it.  Do all tests fail or only some?  I don't like using the configuration system to exclude test assemblies:  this feels like a place where we could easily accidentally lose coverage.

Reply:
Which tests are failing now? We disabled the previously failing tests with https://github.com/dotnet/runtime/commit/a306b03891e447b6f68fe5e664fce8336567ac88 so I'm wondering what causes more tests to fail now.

Reply:
Some of the tests use System.Diagnostics.Process class. we changed its implementation to PNSE. Hence the tests are failing. i can add the skipOnMono Label for the tests failing tests here

Reply:
The Process dependency comes from this callstack:

```
System.PlatformNotSupportedException : Operation is not supported on this platform.
    at System.Diagnostics.Process..ctor(String machineName, Boolean isRemoteMachine, Int32 processId, ProcessInfo processInfo)
    at System.Diagnostics.Process.GetCurrentProcess()
    at System.Configuration.ClientConfigPaths..ctor(String exePath, Boolean includeUserConfig)
    at System.Configuration.ClientConfigPaths.GetPaths(String exePath, Boolean includeUserConfig)
    at System.Configuration.ClientConfigurationHost.get_ConfigPaths()
    at System.Configuration.ClientConfigurationHost.GetStreamName(String configPath)
    at System.Configuration.ClientConfigurationHost.get_IsAppConfigHttp()
    at System.Configuration.Internal.DelegatingConfigHost.get_IsAppConfigHttp()
    at System.Configuration.ClientConfigurationSystem..ctor()
    at System.Configuration.ConfigurationManager.EnsureConfigurationSystem()
```

@Anipik I tried catching the PNSE in `ClientConfigPaths` and that allowed the System.Runtime.Caching tests to work again. I think it's worth doing that?

Reply:
@akoeplinger it will be functional change, we can fix that in a follow up change. i would also ask the area owners if catching the exception is the right fix here, it might have some side effects.

Reply:
Interesting, this is a new failure point this release https://github.com/dotnet/runtime/commit/40d4b206233fcca34b3c8774c418668285b8f1a4.  Might be good to be resilient to no Process support and just fall through.  /cc @eerhardt

Reply:
I think it would be fine to just check `if (!RuntimeInformation.IsOSPlatform(OSPlatform.Browser))` before calling `Process.GetCurrentProcess()` in `System.Configuration.ClientConfigPaths..ctor(String exePath, Boolean includeUserConfig)`. In the Browser case, just leaving `ApplicationUri` empty will be fine.

Reply:
@eerhardt configurationManager builds for netstandard2.0. OSPlatform.Browser is not exposed on netstandard, should we add this to netstandard.dll or have a different net5.0 implementation for configuration manager ?

Reply:
`if (!RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))` will do the trick in that case.

Reply:
I will fix it in a follow up pr.